platform/ios/src/MGLAnnotationView.h

```diff
@@ -16,6 +18,14 @@ NS_ASSUME_NONNULL_BEGIN
 - (instancetype)initWithReuseIdentifier:(nullable NSString *)reuseIdentifier;
 
 /**
+ This property will be set to the associated annotation when the view is visible.
+ 
+ When the view is queued and waiting to be reused, the value will be set to nil.
+ This property should not be changed directly.
+ */
+@property (nonatomic, strong, nullable) id <MGLAnnotation> annotation;
```

Comment:
The annotation corresponds to the visibility of the view so it could be useful to override in order to animate the adding of the annotations views. However, I'd also prefer to make this property readonly and maybe expose `-mapView:didAddAnnotationViews:`